open-api/rest-catalog-open-api.yaml

```diff
@@ -3324,6 +3324,279 @@ components:
+          description: "List of integer column ids for each corresponding value"
+        values:
+          type: array
+          items:
+            $ref: '#/components/schemas/BinaryTypeValue'
+          description: "List of Binary values encoded as hexadecimal strings, matched to 'keys' by index"
+      example:
+        {
+            "keys": [ 3,4 ],
+            "values": [ "05000000","0A000000" ]
+        }
+
+    PrimitiveTypeValue:
+      oneOf:
+        - $ref: '#/components/schemas/BooleanTypeValue'
+        - $ref: '#/components/schemas/IntegerTypeValue'
+        - $ref: '#/components/schemas/LongTypeValue'
+        - $ref: '#/components/schemas/FloatTypeValue'
+        - $ref: '#/components/schemas/DoubleTypeValue'
+        - $ref: '#/components/schemas/DecimalTypeValue'
+        - $ref: '#/components/schemas/StringTypeValue'
+        - $ref: '#/components/schemas/UUIDTypeValue'
+        - $ref: '#/components/schemas/DateTypeValue'
+        - $ref: '#/components/schemas/TimeTypeValue'
+        - $ref: '#/components/schemas/TimestampTypeValue'
+        - $ref: '#/components/schemas/TimestampTzTypeValue'
+        - $ref: '#/components/schemas/TimestampNanoTypeValue'
+        - $ref: '#/components/schemas/TimestampTzNanoTypeValue'
+        - $ref: '#/components/schemas/FixedTypeValue'
+        - $ref: '#/components/schemas/BinaryTypeValue'
+
+    FileContent:
+      type: string
+      enum:
+        - data
+        - position-deletes
+        - equality-deletes
+
+    FileFormat:
+      type: string
+      enum:
+        - avro
+        - orc
+        - parquet
+
+    ContentFile:
+      type: object
+      required:
+        - spec-id
+        - content
+        - file-path
+        - file-format
+        - file-size-in-bytes
+        - record-count
+      properties:
+        content:
+          $ref: '#/components/schemas/FileContent'
+        file-path:
+          type: string
+        file-format:
+          $ref: '#/components/schemas/FileFormat'
+        spec-id:
+          type: integer
+        partition:
+          $ref: '#/components/schemas/PartitionData'
+        file-size-in-bytes:
+          type: integer
+          format: int64
+          description: "Total file size in bytes"
+        record-count:
+          type: integer
+          format: int64
+          description: "Number of records in the file"
+        key-metadata:
+          allOf:
+            - $ref: '#/components/schemas/BinaryTypeValue'
+          description: "Encryption key metadata blob"
+        split-offsets:
+          type: array
+          items:
+            type: integer
+            format: int64
+          description: "List of splittable offsets"
+        sort-order-id:
+          type: integer
+
+    DataFile:
+      allOf:
+        - $ref: '#/components/schemas/ContentFile'
+      type: object
+      properties:
+        column-sizes:
+          allOf:
+            - $ref: '#/components/schemas/CountMap'
```

Comment:
I think these stats fields are direct $ref instead of allOf

Reply:
According to the OpenAPI docs for elements using the ref field, all other fields are ignored. Using allOf is a workaround for that. However, this is just for the descriptions which have no effect on functionality so it can be removed. 

Issue: https://github.com/OAI/OpenAPI-Specification/issues/1514

Reply:
I am fine either way, let's see what others think.

Reply:
Looks like elsewhere we use `$ref` directly. Does it not generate code correctly?

I'm looking at `AddViewVersionUpdate`, which has `view-version` that is a `$ref` to `#/components/schemas/ViewVersion`. That looks fine in [python](https://github.com/apache/iceberg/pull/9717/files#diff-ffc6f59bffb61af69bc4ef19c7b28c9fefe015b2716b35db1deb6a5a884ca4feL331-L333) and `action` is not dropped.

I think it's fine either way. Whatever is most compatible.

Reply:
Yeah I'm fine either way. This works for the python client fine. But when loading the rest yaml into a swagger editor the descriptions aren't persisted. For instance, the `CommitTableRequest` model has this

Current:
<img width="458" alt="Screenshot 2024-02-26 at 2 41 29 PM" src="https://github.com/apache/iceberg/assets/19402187/aabcf585-3b99-4e97-ad62-285363ad07b7">


After adding this logic:
<img width="729" alt="image" src="https://github.com/apache/iceberg/assets/19402187/bdb07cad-c9ac-4110-bbb4-4275328477ac">

Furthermore, this issue falls to generating the clients as well like javascript:

Before:
```
/**
 * @member {module:model/TableIdentifier} identifier
 */
CommitTableRequest.prototype.identifier = undefined;
```

After:
```
/**
 * Table identifier to update; must be present for CommitTransactionRequest
 * @member {Object} identifier
 */
CommitTableRequest.prototype.identifier = undefined;
```

Reply:
Ah, good catch. Then we should probably update other cases after this PR. Thanks, @geruh!